hail/python/hailtop/hailctl/config/cli.py

```diff
@@ -9,6 +9,7 @@
 import typer
 from typer import Argument as Arg
 
+from hailtop.batch_client.parse import CPU_REGEXPAT, MEMORY_REGEXPAT
```

Comment:
Do you know how shell autocomplete is actually implemented? Is it just repeatedly invoking the `hailctl` executable or is there some long-lived `hailctl` process and it's communicating with the shell? This would be useful information to know for making the autocomplete snappier in the future as it's a bit laggy at the moment.

Reply:
It's much snappier when we don't import the regex patterns for cpu and memory as well as the RouterAsyncFS.

Reply:
Let me know what you'd like to do here.

Reply:
So those imports are only needed to execute the validation lambdas, which aren't necessary for autocomplete. If we go with my other suggestion about turning these tuples into classes, and instead of those lambdas have a `validate` method on the classes, I think we should get best of both worlds because those imports can happen inside the `validate` method.